docs/DeveloperGuide.md

```diff
@@ -1,38 +1,327 @@
+
+This command extends the Command class and implements the following operations:
+
+`UserDetailsCommand#execute()` — Interacts with the user via the UI to gather personal information and saves it using the `Storage` class.
+Personal details such as name, age, and gender are stored in a `UserDetails` object, which is then serialized to a file by `Storage`.
+These operations are exposed in the system as follows:
+
+The `UserDetailsCommand` is instantiated and called when the user chooses to enter or update their personal details.
+The execute method of this command interacts with other components like `Ui` for input/output and `Storage` for persisting user data.
+Below is the sequence of actions performed by the `UserDetailsCommand` to collect user information.
+
+![User Details Collection Sequence Diagram](images/UserDetailsCommandSequenceDiagram.png)
+
+Each step of the execute method interacts with the user to collect a specific piece of information. Conditional logic is applied to request the anniversary date if the relationship status warrants it. After all information is gathered, the UserDetails object's properties are set accordingly. The final step involves the Storage component saving the details, thus persisting the data for future sessions.
+
+The above sequence diagram depicts the interaction between UserDetailsCommand, the UI, and the Storage component, which highlights the flow of data collection and storage.
+
+
+### [Implemented] History Tracking feature
+
+#### Implementation
+
+The existing History Tracking feature keeps a database of previously visited restaurants, activities and gifts. It extends the Command class with a new ```ViewHistoryCommand```. Additionally, it implements the following operation:
+- ```execute(FavouritesList favourites, FoodList foods, ActivityList activities, Ui ui, Storage storage, UserDetails userDetails, GiftList gifts)``` — Displays a list of all past restaurants, activities and gifts based on the user's past history.
+
+These operations are detailed in the ```Parser``` class, where the ```parseCommand(String userInput, UserDetails userDetails)``` method is updated to return a ```ViewHistoryCommand``` when the `"history"` command is entered by the user.
+
+Given below is an example usage scenario of how the History Tracking mechanism behaves at each step.
+
+Step 1. The user launches the application and executes the `idea`/`itinerary`/`smart`/`gift` command. The relevant command is parsed by the `parseCommand` method in the `Parser` class, which creates a `GenerateIdeaCommand`/ `GenerateItineraryCommand`/`GenerateSmartItineraryCommand`/`GenerateGiftCommand` instance
```

Comment:
Its good to have this much detail! But I do think it might be a bit excessive to go through the geneate command here again?